pkg/metrics/metrics.go

```diff
@@ -1414,9 +1415,9 @@ func NewBPFMapPressureGauge(mapname string, threshold float64) *GaugeWithThresho
 }
 
 func init() {
-	MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{Namespace: Namespace}))
-	// TODO: Figure out how to put this into a Namespace
-	// MustRegister(prometheus.NewGoCollector())
+	MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{Namespace: Namespace}))
+	MustRegister(collectors.NewGoCollector())
+	MustRegister(collectors.NewBuildInfoCollector())
```

Comment:
In the past, I was struggling to have meaningful label value for this collector, do you know how we can resolve it ? The related docs suggested a way to build, but I don't think it will fit well with our build process.

Please also find below the output from my local installation for this branch.

> This is usually accomplished by building from outside of GOPATH, specifying the full address of the main package, e.g. "GO111MODULE=on go run github.com/prometheus/client_golang/examples/random". If built without Go module support, all label values will be "unknown". If built with Go module support but using the source code from the local file system, the "path" will be set appropriately, but "checksum" will be empty and "version" will be "(devel)".

```
$ ksysex ds/cilium -- cilium metrics list -p "go_build_info"
Defaulted container "cilium-agent" out of: cilium-agent, ebpf-mount (init), clean-cilium-state (init)
Metric          Labels                                                          Value
go_build_info   checksum="" path="github.com/cilium/cilium" version="(devel)"   1.000000
```

Reply:
Ah, I believe normally it will pull the version from the `main.version` if you're building with ldflags. I'm not sure about how it works with go modules TBH, I haven't written much Go since right before modules were added. 

I also thought this metric included the go version but I think that's actually from the `go_info` metric provided by the `GoCollector`.

I'd be fine with omitting the build info collector if we don't think it will be useful, or if it will require additional work to make it provide useful information.

Reply:
considered that it has only 1 metric with two labels checksum and version, I think we can drop this collector.

Go 1.18 is having below runtime buildinfo method, surely we can leverage it in near future.

> The new [debug/buildinfo](https://tip.golang.org/pkg/debug/buildinfo) package provides access to module versions, version control information, and build flags embedded in executable files built by the go command. The same information is also available via [runtime/debug.ReadBuildInfo](https://tip.golang.org/pkg/runtime/debug#ReadBuildInfo) for the currently running binary and via go version -m on the command line.

Reply:
Done